src/api.c

```diff
@@ -232,7 +236,7 @@ int cg_chmod_path(const char *path, mode_t mode, int owner_is_umask)
 		mask = umask|gmask|omask|S_ISUID|S_ISGID|S_ISVTX;
 	}
 
-	if (chmod(path, mode & mask))
+	if (fchmod(fd, mode & mask))
 		goto fail;
 
 	return 0;
```

Comment:
Don't you need to free the `fd` prior to this return as well?

Reply:
@drakenclimber Thanks for the review. Yes, it needs to closed.